recipes-soletta/soletta/soletta_git.bb

```diff
@@ -81,7 +86,55 @@ do_configure_prepend() {
    export TARGETAR="${AR}"
 }
 
+
 do_compile() {
+   # changing the home directory to the working directory, the .npmrc will be created in this directory
+   export HOME=${WORKDIR}
+
+   # does not build dev packages
+   npm config set dev false
+
+   # access npm registry using http
+   npm set strict-ssl false
+   npm config set registry http://registry.npmjs.org/
+
+   # configure http proxy if neccessary
+   if [ -n "${http_proxy}" ]; then
+       npm config set proxy ${http_proxy}
+       NODE_GYP_PROXY="--proxy=${http_proxy}"
+   fi
+   if [ -n "${HTTP_PROXY}" ]; then
+       npm config set proxy ${HTTP_PROXY}
+       NODE_GYP_PROXY="--proxy=${HTTP_PROXY}"
+   fi
+
+   # configure cache to be in working directory
+   npm set cache ${WORKDIR}/npm_cache
+
+   # clear local cache prior to each compile
+   npm cache clear
+
+   case ${TARGET_ARCH} in
+       i?86) targetArch="ia32"
+           ;;
+       x86_64) targetArch="x64"
+           ;;
+       arm) targetArch="arm"
+           ;;
+       mips) targetArch="mips"
+           ;;
+       sparc) targetArch="sparc"
+           ;;
+       *) echo "unknown architecture"
+          exit 1
+           ;;
+   esac
+
+   # Export needed variables to build Node.js bindings
+   export CFLAGS="$CFLAGS -fPIC"
+   export CXXFLAGS="$CXXFLAGS -fPIC"
+   export NODE_GYP="${STAGING_DIR_NATIVE}/${libdir}/node_modules/npm/bin/node-gyp-bin/node-gyp --arch=${targetArch} ${NODE_GYP_PROXY}"
```

Comment:
Good catch. Forgot to commit a local change that I made to use TARGET instead of NATIVE here.